syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.petadoption";
option java_outer_classname = "PetAdoptionProto";

package petadoption;

service PetAdoptionService {
    // Register a pet.
    rpc RegisterPet (PetInfo) returns (RegistrationResponse);
    // Search for a pet.
    rpc SearchPet (SearchRequest) returns (PetList);
}

message PetInfo {
    string name = 1;
    string gender = 2;
    int32 age = 3;
    string breed = 4;
    bytes picture = 5;
}

message RegistrationResponse {
    bool success = 1;
    string message = 2;
}

message SearchRequest {
    string query = 1;
}

message PetList {
    repeated PetInfo pets = 1;
}